### PR TITLE
Add Docker Swarm configuration example

### PR DESCRIPTION
**Pull request by roidelapluie**
Signed-off-by: Julien Pivotto <roidelapluie@inuits.eu>

### CODE REVIEW

File: documentation/examples/prometheus-dockerswarm.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
+# A example scrape configuration for running Prometheus with
+# Docker Swarm.
+#
+# This exemple works with Prometheus node_exporter deployed with:
```
Comment (brian-brazil) [inline documentation/examples/prometheus-dockerswarm.yml:4]: example

File: documentation/examples/prometheus-dockerswarm.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+# docker service create --name node -l prometheus-job=node \
```
Comment (brian-brazil) [inline documentation/examples/prometheus-dockerswarm.yml:5]: Hmm, do we want to be encouraging people to run the node exporter in Docker? @SuperQ

File: documentation/examples/prometheus-dockerswarm.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    relabel_configs:
+      # Only keep containers that should be running.
+      - source_labels: [__meta_dockerswarm_task_desired_state]
+        regex: running
+        action: keep
+      # Only keep containers that have a `prometheus-job` label.
+      - source_labels: [__meta_dockerswarm_service_label_prometheus_job]
+        regex: .+
+        action: keep
+      # Use the tasks labels that are prefixed by `prometheus-`.
```
Comment (brian-brazil) [inline documentation/examples/prometheus-dockerswarm.yml:32]: task

File: documentation/examples/prometheus-dockerswarm.yml (hunk 2 above)
Comment (SuperQ) [reply documentation/examples/prometheus-dockerswarm.yml:5]: Similar to Kubernetes, for swarm users, this is likely to be common. Having correct documentation examples here is probably better than many of the incorrect docker-compose.yml configs I've seen from users having trouble with it.

Comment (SuperQ) [reply documentation/examples/prometheus-dockerswarm.yml:5]: I did a bit more digging, and it looks like the big problem is that Swarm mode doesn't support `--pid=host`, so I'm pretty sure this won't work correctly. :frowning:

Comment (roidelapluie) [reply documentation/examples/prometheus-dockerswarm.yml]: updated for cadvisor

**Comment by brian-brazil** [comment]
:+1:

**Comment by SuperQ** [comment]
Nice!

**Pull request merged**